src/PHPStan/Collector/ContextCollector.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace TwigStan\PHPStan\Collector;
+
+use PhpParser\Node;
+use PHPStan\Collectors\Collector;
+
+/**
+ * @phpstan-type TemplateData = array{
+ *      startLine: int,
+ *      endLine: int,
+ *      template: string,
+ *      context: string,
+ *  }
+ *
+ * @template-covariant TNodeType of Node
+ * @implements Collector<TNodeType, array<TemplateData>>
+ */
+interface ContextCollector extends Collector
```

Comment:
It's simpler for the implementation if ALL the collector have the same signature. `array<TemplateData>`